simulate_groups.py

```diff
@@ -76,7 +80,10 @@ def simulate_ll(n,
         num_groups = len(groups)
         if group_sparsity < 1.0:
             num_groups_to_keep = int(num_groups * group_sparsity)
-            groups_to_keep = np.random.choice(num_groups, num_groups_to_keep)
+            if prev_groups is None:
+                groups_to_keep = np.random.choice(num_groups, num_groups_to_keep, replace=False)
+            else:
+                groups_to_keep = np.random.choice(prev_groups, num_groups_to_keep, replace=False)
```

Comment:
So before, `group_sparsity` specified the proportion of all the groups that will have nonzero coefficients. When `groups_to_keep` is passed in, `group_sparsity` now becomes the proportion of the previous groups that will have nonzero coefficients, right?

Does this mean any new groups that aren't in `prev_groups` will still have zero coefficients (in other words, the nonzero groups when `prev_groups` are passed in will be a subset of the nonzero groups from the first run)?

Just trying to wrap my head around exactly what this is doing.

Reply:
`group_sparsity` should still be the proportion of all groups. If there's ten groups total, and in the first run `group_sparsity` is 0.8, then 8 groups  (say all but 9 and 10) will have nonzero coefficients. If the next run has `group_sparsity=0.4` and `prev_groups` passed in, it will have 4 groups with nonzero coefficients, but it will be constrained to pick those four groups from the 8 previous (prevented from picking 9 or 10 as one of the four). There's probably some other way to do this with the beta values that are already passed in, but this seemed more straightforward to me.

Reply:
So what would happen if you had `group_sparsity=0.9` on the second run in your example? Would it just pick all of the previous groups and groups 9 and 10 would still have zero coefficients?

(I know this is nitpicky, I just want to make sure edge cases get documented in case we use this more in the future)

Reply:
Also I think "sparsity" usually refers to the percentage of zeroes, and I just noticed that here it's the opposite (percentage of non-zero values) so we should probably change the name. I can do that in a separate PR, though.

Reply:
Re. [my previous comment](https://github.com/greenelab/simulate-groups/pull/2/files#r656501484), I tested the case when the second run has a higher sparsity than the first run, and it throws an error. So I think what we want to do is this:
```python
num_groups = len(groups)
if group_sparsity < 1.0:
    if prev_groups is None:
        num_groups_to_keep = int(num_groups * group_sparsity)
        groups_to_keep = np.random.choice(num_groups, num_groups_to_keep, replace=False)
    else:
        num_groups_to_keep = int(prev_groups * group_sparsity)
        groups_to_keep = np.random.choice(prev_groups, num_groups_to_keep, replace=False)
```
Either you can make this change and regenerate the data, or I can make this change in a later PR (if this would make you have to rerun wenda again, I'm happy to stick with what we have for now and create an issue for the future). Up to you.

Reply:
Created #3 - I'll merge this now and make the change in the future.